src/document-processor.ts

```diff
@@ -0,0 +1,400 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {Program} from 'estree';
+import * as jsc from 'jscodeshift';
+import * as parse5 from 'parse5';
+import {Document, Import, isPositionInsideRange, Severity, Warning} from 'polymer-analyzer';
+import * as recast from 'recast';
+
+import {ConversionSettings} from './conversion-settings';
+import {attachCommentsToFirstStatement, canDomModuleBeInlined, createDomNodeInsertStatements, filterClone, getCommentsBetween, getNodePathInProgram, insertStatementsIntoProgramBody, serializeNodeToTemplateLiteral} from './document-util';
+import {NamespaceMemberToExport} from './js-module';
+import {removeNamespaceInitializers} from './passes/remove-namespace-initializers';
+import {removeToplevelUseStrict} from './passes/remove-toplevel-use-strict';
+import {removeUnnecessaryEventListeners} from './passes/remove-unnecessary-waits';
+import {removeWrappingIIFEs} from './passes/remove-wrapping-iife';
+import {rewriteToplevelThis} from './passes/rewrite-toplevel-this';
+import {ConvertedDocumentFilePath, ConvertedDocumentUrl, OriginalDocumentUrl} from './urls/types';
+import {UrlHandler} from './urls/url-handler';
+import {isOriginalDocumentUrlFormat} from './urls/util';
+import {replaceHtmlExtensionIfFound} from './urls/util';
+
+/**
+ * Keep a set of elements to ignore when Recreating HTML contents by adding
+ * code to the top of a program.
+ */
+const generatedElementBlacklist = new Set<string|undefined>([
+  'base',
+  'link',
+  'meta',
+  'script',
+]);
+
+
+
+export type ScanResult =
+    JsModuleScanResult|DeleteFileScanResult|HtmlDocumentScanResult;
+/**
+ * Contains information about how an existing file should be converted to a new
+ * JS Module. Includes a mapping of its new exports.
+ */
+export interface JsModuleScanResult {
+  type: 'js-module';
+  originalUrl: OriginalDocumentUrl;
+  convertedUrl: ConvertedDocumentUrl;
+  convertedFilePath: ConvertedDocumentFilePath;
+  exportMigrationRecords: NamespaceMemberToExport[];
+}
+
+/**
+ * Contains information that an existing file should be deleted during
+ * conversion.
+ */
+export interface DeleteFileScanResult {
+  type: 'delete-file';
+  originalUrl: OriginalDocumentUrl;
+  convertedUrl: undefined;
+  convertedFilePath: undefined;
+}
+
+/**
+ * Contains information that an existing file should be converted as a top-level
+ * HTML file (and not as a new JS module).
+ */
+export interface HtmlDocumentScanResult {
```

Comment:
move these types to `document-scanner.ts`

Reply:
done